app/edit-ui/src/components/TextInput/TextInput.jsx

```diff
@@ -0,0 +1,98 @@
+import React, {useCallback, useEffect, useRef} from 'react'
+import classnames from '../../util/classnames'
+// import PropTypes from 'prop-types'
+import styles from './TextInput.css'
+
+// TextInput.propTypes = {}
+
+export default React.forwardRef(function TextInput(
+  {
+    accent,
+    autoresize,
+    className,
+    error,
+    multiline,
+    inLabel: _,
+    readOnly,
+    resizable,
+    ...props
+  },
+  ref
+) {
+  if (!ref) {
+    ref = useRef()
+  }
+
+  const value = props.value === null ? '' : props.value
+
+  if (error) {
+    accent = 'error'
+  }
+
+  // Resizing section
+
+  const maybeResize = useCallback(() => {
+    if (autoresize && !resizable && ref.current) {
+      ref.current.style.height = 'auto'
+      ref.current.style.height = `${ref.current.scrollHeight + 2}px`
+    }
+  }, [autoresize, ref, resizable])
+
+  useEffect(maybeResize, [value])
+
+  // Cursor maintaining section
```

Comment:
I really, really want to get to the bottom of why we need this in the first place.

Reply:
I'll be glad to look into it! I did want to avoid this logic in the new TextInput component and fix it in Publish instead, but I wanted to get some visible results first, before you leave